code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -467,3 +472,32 @@
 								break
 						beaker.reagents.add_reagent(r_id, min(O.reagent_contents[r_id], space))
 						remove_object(O)
+
+/obj/machinery/reagentgrinder/proc/mix()
+
+		//For butter and other things that would change upon shaking or mixing
+		power_change()
+		if(stat & (NOPOWER|BROKEN))
+				return
+		if (!beaker)
+				return
+		playsound(src.loc, 'sound/machines/juicer.ogg', 20, 1)
+		var/offset = prob(50) ? -2 : 2
+		animate(src, pixel_x = pixel_x + offset, time = 0.2, loop = 250) //start shaking
+		operating = 1
+		updateUsrDialog()
+		spawn(50)
+				pixel_x = initial(pixel_x) //return to its spot after shaking
+				operating = 0
+				updateUsrDialog()
+				if (beaker.reagents.total_volume)
+						//Recipe to make Butter
+						if (beaker.reagents.has_reagent("milk", 15))
```

Comment:
This is the only part of this that is actually my code. The rest of this is changes you're asking me to make to pre-existing grinder code.